apps/smp-server/Main.hs

```diff
@@ -94,6 +104,11 @@ main = do
     ServerDelete -> do
       deleteServer
       putStrLn "Server key, config file and store log deleted"
+  where
+    checkPubkeyAlgorihtm :: String -> IO ()
+    checkPubkeyAlgorihtm alg
+      | alg == "ED448" || alg == "ED25519" = pure ()
+      | otherwise = error "unsupported public-key algorithm"
```

Comment:
there is a better function for exit we use other than error that simply crashes

```suggestion
      | otherwise = putStrLn ("unsupported public-key algorithm " <> alg) >> exitFailure
```